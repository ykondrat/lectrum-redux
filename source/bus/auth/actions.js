// Types
import { types } from './types';

// Async
export const signupAsync = (userData) => ({
    type: types.SIGNUP_ASYNC,
    payload: userData
});

export const loginAsync = (credentials) => ({
    type: types.LOGIN_ASYNC,
    payload: credentials
});

export const authenticateAsync = (credentials) => ({
    type: types.AUTHENTICATE_ASYNC,
    payload: credentials
});

export const initializeAsync = () => ({
    type: types.INITIALIZE_ASYNC
});

export const logoutAsync = () => ({
    type: types.LOGOUT_ASYNC
});

// Sync
export const authenticate = () => ({
    type: types.AUTHENTICATE
});

export const initialize = () => ({
    type: types.INITIALIZE
});

export const logout = () => ({
    type: types.LOGOUT
});
