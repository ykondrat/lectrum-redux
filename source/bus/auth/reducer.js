// Core
import { Map } from 'immutable';

// Types
import { types } from './types';

const initialState = Map({
    isAuthenticated: false,
    isInitialized: false
});

export const authReducer = (state = initialState, action) => {

    switch (action.type) {
        case types.AUTHENTICATE:
            return state.set('isAuthenticated', true);
        case types.INITIALIZE:
            return state.set('isInitialized', true);
        case types.LOGOUT:
            return state.set('isAuthenticated', false);
        default:
            return state;
    }

}
