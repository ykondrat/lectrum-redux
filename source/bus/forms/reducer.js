// Core
import { combineForms } from 'react-redux-form';

export const formsReducer = combineForms({
    user: {
        profile: {
            firstName: '',
            lastName: '',
            avatar: []
        }
    }
}, 'forms');
