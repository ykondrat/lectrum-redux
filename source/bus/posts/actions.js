// Types
import { types } from './types';

// Async
export const fetchPostsAsync = () => ({
    type: types.FETCH_POSTS_ASYNC
});

// Sync
export const fillPosts = (posts) => ({
    type: types.FILL_POSTS,
    payload: posts
});
