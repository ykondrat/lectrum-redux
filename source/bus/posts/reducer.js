// Core
import { List, fromJS } from 'immutable';

// Types
import { types } from './types';

const initialState = List();

const findPostIndex = (state, postId) => {
    const index = state.findIndex((post) => post.get('id') === postId);

    return index;
}

export const postsReducer = (state = initialState, action) => {

    switch (action.type) {
        case types.FILL_POSTS:
            return fromJS(action.payload);

        case types.LIKE_POST:
            return state.updateIn(
                [findPostIndex(state, action.payload.postId), 'likes'],
                (likes) => likes.unshift(action.payload.liker)
            );

        case types.UNLIKE_POST:
            return state.updateIn(
                [findPostIndex(state, action.payload.postId), 'likes'],
                (likes) => likes.filter((liker) => liker.get('id') !== action.payload.userId)
            );

        case types.CREATE_POST:
            return state.unshift(fromJS(action.payload));

        case types.REMOVE_POST:
            return state.remove(findPostIndex(state, action.payload));

        default:
            return state;
    }

}
