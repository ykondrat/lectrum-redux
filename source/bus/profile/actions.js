// Types
import { types } from './types';

// Async


// Sync
export const clearProfile = () => ({
    type: types.CLEAR_PROFILE
});

export const fillProfile = (profile) => ({
    type: types.FILL_PROFILE,
    payload: profile
});
