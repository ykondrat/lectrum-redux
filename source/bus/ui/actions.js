// Types
import { types } from './types';

// Sync
export const startFetching = () => ({
    type: types.START_FETCHING
});

export const stopFetching = () => ({
    type: types.STOP_FETCHING
});
