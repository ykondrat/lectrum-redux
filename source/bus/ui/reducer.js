// Core
import { Map } from 'immutable';

// Types
import { types } from './types';

const initialState = Map({
    isFetching: false,
    isOnline: false
});

export const uiReducer = (state = initialState, action) => {
    switch (action.type) {
        case types.START_FETCHING:
            return state.set('isFetching', true);
        case types.STOP_FETCHING:
            return state.set('isFetching', false);
        case types.SET_ONLINE_STATE:
            return state.set('isOnline', true);
        case types.SET_OFFLINE_STATE:
            return state.set('isOnline', false);
        default:
            return state;
    }
}
