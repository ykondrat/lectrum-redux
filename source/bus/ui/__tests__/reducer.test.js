// Reducer
import { uiReducer } from '../reducer';

import * as actions from '../actions';

const actionTest = {
    type: 'INIT_TEST_DEFAULT'
}

describe('uiReducer tests:', () => {
    test('uiReducer test for default switch case', () => {
        expect(
            uiReducer( void 0, actionTest)
        ).toMatchSnapshot();
    });

    test('uiReducer test for startFetching', () => {
        expect(
            uiReducer(void 0, actions.startFetching())
        ).toMatchSnapshot();
    });

    test('uiReducer test for stopFetching', () => {
        expect(
            uiReducer(void 0, actions.stopFetching())
        ).toMatchSnapshot();
    });

    test('uiReducer test for setOnlineState', () => {
        expect(
            uiReducer(void 0, actions.setOnlineState())
        ).toMatchSnapshot();
    });

    test('uiReducer test for setOfflineState', () => {
        expect(
            uiReducer(void 0, actions.setOfflineState())
        ).toMatchSnapshot();
    });

});
