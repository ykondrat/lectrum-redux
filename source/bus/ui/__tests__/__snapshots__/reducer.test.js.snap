// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiReducer tests: uiReducer test for default switch case 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;

exports[`uiReducer tests: uiReducer test for setOfflineState 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;

exports[`uiReducer tests: uiReducer test for setOnlineState 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": true,
}
`;

exports[`uiReducer tests: uiReducer test for startFetching 1`] = `
Immutable.Map {
  "isFetching": true,
  "isOnline": false,
}
`;

exports[`uiReducer tests: uiReducer test for stopFetching 1`] = `
Immutable.Map {
  "isFetching": false,
  "isOnline": false,
}
`;
